components/StyledInputAmount.js

```diff
@@ -43,30 +45,68 @@ const getError = (curVal, minAmount, required) => {
   return Boolean((required && isNil(curVal)) || (minAmount && curVal < minAmount));
 };
 
+const generateCurrencyOptions = availableCurrencies => {
+  return Object.keys(availableCurrencies).map(key => {
+    const availableCurrency = availableCurrencies[key];
+    return {
+      value: key,
+      label: (
+        <Flex fontSize="14px" lineHeight="20px" fontWeight="500" title={availableCurrency.fullName}>
+          <Span>
+            <CurrencyFlag currency={key} size="sm" />
+          </Span>
+          &nbsp;
+          <Span whiteSpace="nowrap" ml={1}>
+            <Span color="black.800">{key}</Span>
+            {` `}
+            <Span color="black.500">({availableCurrency.fullName})</Span>
```

Comment:
With the right helper, this could be as simple as:

```suggestion
            <Span color="black.500">({i18nCurrencyName(intl, currency)})</Span>
```

And no need to pass a complex map for `avalaibleCurrencies`.